src/integrant/repl.clj

```diff
@@ -29,59 +32,83 @@
              {:init-exception %1}
              %2)))
 
-(defn- resume-system [config system]
+(defn- resume-system [config system keys]
   (build-system
-   #(ig/resume config system)
+   #(ig/resume config system (get-keys system keys))
    #(ex-info "Config failed to resume; also failed to halt failed system"
              {:resume-exception %1}
              %2)))
 
-(defn init []
-  (alter-var-root #'state/system (fn [sys]
-                                   (halt-system sys)
-                                   (init-system state/config)))
-  :initiated)
-
-(defn go []
-  (prep)
-  (init))
-
-(defn clear []
-  (alter-var-root #'state/system (fn [sys] (halt-system sys) nil))
-  (alter-var-root #'state/config (constantly nil))
-  :cleared)
-
-(defn halt []
-  (halt-system state/system)
-  (alter-var-root #'state/system (constantly nil))
-  :halted)
-
-(defn suspend []
-  (when state/system (ig/suspend! state/system))
-  :suspended)
-
-(defn resume []
-  (if-let [prep state/preparer]
-    (let [cfg (prep)]
-      (alter-var-root #'state/config (constantly cfg))
-      (alter-var-root #'state/system (fn [sys]
-                                       (if sys
-                                         (resume-system cfg sys)
-                                         (init-system cfg))))
-      :resumed)
-    (throw (prep-error))))
-
-(defn reset []
-  (suspend)
-  (repl/refresh :after 'integrant.repl/resume))
-
-(defn reset-all []
-  (suspend)
-  (repl/refresh-all :after 'integrant.repl/resume))
+(defn init
+  ([] (init nil))
+  ([keys]
+   (alter-var-root #'state/system (fn [sys]
+                                    (halt-system sys keys)
+                                    (init-system state/config keys)))
+   :initiated))
+
+(defn go
+  ([] (go nil))
+  ([keys]
+   (prep)
+   (init keys)))
+
+(defn clear
+  ([] (clear nil))
+  ([keys]
+   (alter-var-root #'state/system (fn [sys] (halt-system sys keys)) nil)
+   (alter-var-root #'state/config (constantly nil))
+   :cleared))
+
+(defn halt
+  ([] (halt nil))
+  ([keys]
+   (halt-system state/system keys)
+   (alter-var-root #'state/system (constantly nil))
+   :halted))
+
+(defn suspend
+  ([] (suspend nil))
+  ([keys]
+   (when state/system (ig/suspend! state/system keys))
+   :suspended))
+
+(defn resume
+  ([] (resume nil))
+  ([keys]
+   (if-let [prep state/preparer]
+     (let [cfg  (prep)
+           keys (or state/initiated-keys keys)]
+       (alter-var-root #'state/config (constantly cfg))
+       (alter-var-root #'state/system (fn [sys]
+                                        (if sys
+                                          (resume-system cfg sys keys)
+                                          (init-system cfg keys))))
+       :resumed)
+     (throw (prep-error)))))
+
+(defn- set-selected-keys! [keys]
+  (alter-var-root #'state/initiated-keys (constantly keys)))
+
+(defn reset
+  ([] (reset nil))
+  ([keys]
+   (set-selected-keys! keys)
```

Comment:
Should this be in `reset`, or should we have something like `set-suspended-keys`?

For that matter, do we need the var at all? Surely `(keys system)` would amount to the same thing?